include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -856,8 +856,13 @@ struct __write_to_idx_if
     void
     operator()(_OutRng&& __out, _SizeType __idx, const ValueType& __v) const
     {
+        // Use of an explicit cast to our internal tuple type is required to resolve conversion issues between our
+        // internal tuple and std::tuple. If the underlying type is not a tuple, then the type will just be passed through.
```

Comment:
I think we may need a similar fix on line 808.

Reply:
We don't have any tests catching it here, but I agree the issue may arise so I have added a similar fix.